packages/x-data-grid/src/tests/rowSelection.DataGrid.test.tsx

```diff
@@ -839,6 +839,34 @@ describe('<DataGrid /> - Row selection', () => {
       expect(onRowSelectionModelChange.lastCall.args[0]).to.deep.equal(includeRowSelection([]));
     });
 
+    it('should call onRowSelectionModelChange with the correct reason when clicking on a row', async () => {
+      const onRowSelectionModelChange = spy();
+      const { user } = render(
+        <TestDataGridSelection
+          checkboxSelection
+          pagination
+          onRowSelectionModelChange={onRowSelectionModelChange}
+        />,
+      );
+      await user.click(getCell(0, 0).querySelector('input')!);
+      expect(onRowSelectionModelChange.lastCall.args[1].reason).to.deep.equal('singleRowSelection');
```

Comment:
This would be more correct? But maybe there is no meaningful difference. Maybe the error message when it fails is different. 

```suggestion
      expect(onRowSelectionModelChange.lastCall.args[1].reason).to.equal('singleRowSelection');
```

x the other.